src/App.jsx

```diff
@@ -8,26 +8,21 @@ import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
 import 'bootstrap/dist/css/bootstrap.min.css';
 import './styles/App.css';
 import SettingsHeader from "./views/SettingsHeader";
-import LoginProvider from "./util/LoginProvider";
+import requestProvider from './util/requestProvider';
 
 
 export default function () {
     const [cluster, setCluster] = useState("live");
     const [flow, setFlow] = useState("browser");
     const [sessionActive, setSessionActive] = useState(false);
 
-    const clusterChange = (event) => {
-        setCluster(event.target.value);
-    }
+    const clusterChange = (event) => setCluster(event.target.value);
+    const flowChange = (event) => setFlow(event.target.value);
 
-    const flowChange = (event) => {
-        setFlow(event.target.value);
-    }
+    const requests = requestProvider(cluster, flow);
 
     const checkSession = async () => {
-        let login = new LoginProvider(cluster, flow);
-        var jsonRes = await API.makeRequest(login.getSessionRequest());
-
+        const jsonRes = await API.makeRequest(requests.session());
         const isSessionActive = !(jsonRes.status == 401);
```

Comment:
```suggestion
        const isSessionActive = !(jsonRes.status === 401);
```

if there is anything I remember from my time as a nodeJS dev it's that you should always check type equality :D

Reply:
Indeed. Happens in a language that is basically unusable without a linter :D 
I created #7 as a consequence.